### PR TITLE
Release update

### PR DESCRIPTION
**Pull request by ridv**
### Description:
Updating release instructions to reflect the project reboot in Github.

### Testing Done:
Released Aurora 0.24.1 with instructions

### CODE REVIEW

**Comment by ridv** [comment]
@lenhattan86 can you follow the instructions here and see if they work?

**Comment by lenhattan86** [comment]
sure. let me try it out

File: docs/development/committers-guide.md [hunk 1]
```diff
@@ -1,68 +1,49 @@
...
-The following will guide you through the steps to create a release candidate, vote, and finally an
-official Apache Aurora release. Before starting your gpg key should be in the KEYS file and you
-must have access to commit to the dist.a.o repositories.
+The following will guide you through the steps to create a release candidate and an
+official Aurora Scheduler release. Before starting your gpg key should be in the KEYS file.
 
-1. Ensure that all issues resolved for this release candidate are tagged with the correct Fix
-Version in JIRA, the changelog script will use this to generate the CHANGELOG in step #2.
-To assign the fix version:
+1. Ensure that all merged pull requests for this release candidate are tagged with the correct version,
```
Comment (lenhattan86) [inline docs/development/committers-guide.md:43]: It is not clear for me how to verify the tagged versions. For example, if we have a PR merged on top of 0.24.1. should it be tagged with 0.24.1? 
If it is not tagged correctly. How do we fix it?

File: docs/development/committers-guide.md [hunk 2]
```diff
@@ -1,68 +1,49 @@
...
+the changelog script will use this to generate the CHANGELOG in step #2.
+Merged pull requests can be found [here](https://github.com/aurora-scheduler/aurora/pulls?q=is%3Apr+is%3Amerged+).
 
-    * Look up the [previous release date](https://issues.apache.org/jira/browse/aurora/?selectedTab=com.atlassian.jira.jira-projects-plugin:versions-panel).
-    * Query all issues resolved after that release date: `project = AURORA AND status in (resolved, Closed) and fixVersion is empty and resolutiondate >= "YYYY/MM/DD"`
-    * In the upper right corner of the query result, select Tools > Bulk Edit.
-    * Select all issues > edit issue > set 'Change Fix Version/s' to the release version.
-    * Make sure to uncheck 'Send mail for this update' at the bottom.
 
 2. Prepare RELEASE-NOTES.md for the release. This just boils down to removing the "(Not yet
```
Comment (lenhattan86) [inline docs/development/committers-guide.md:48]: line 51 should be 3. etc

**Pull request merged**